src/lower.jl

```diff
@@ -250,6 +250,18 @@ function lower(root::FinchNode, ctx::AbstractCompiler, ::DefaultStyle)
     elseif root.kind === loop
         @assert root.idx.kind === index
         idx_sym = ctx.freshen(root.idx.name)
+        
+        # Generalized Strength Reduction
```

Comment:
This is not strictly correct, it's too eager. There might be more than one access, just because one of them can be reduced does not mean that all of them can.

Reply:
e.g.
```
for i = _
    A[] += B[i]
    C[] += 1
end
```

Reply:
My approach here would be to instead guard the assignment with a `diagmask`.

```
for i = _
    A[] += B[i]
    if diagmask[i, getstop(root.ext)]
        C[] += measure(root.ext)
    end
end
```

Reply:
Then you could also add a few rules about collapsing identical conditions, and perhaps we could make a mask called `once[i]` which is just a `spike(false, true)`

Reply:
Could we perhaps separate issue #181 from this PR?

Reply:
Yea I agree that it would be better handled in separate PR. I'll revert lower.jl.